### PR TITLE
ADD: skip when AutomountServiceAccountToken=false

### PR DESCRIPTION
**Pull request by nirav24**
fixes https://github.com/stackrox/kube-linter/issues/135

### CODE REVIEW

**Review by viswajithiii — COMMENTED** [review]
Thanks for doing this!

We're generally trying to improve our unit test coverage. Would you be open to writing a unit test for this? [This file](https://github.com/stackrox/kube-linter/blob/main/pkg/templates/antiaffinity/template_test.go) should have a pattern you can follow. (Unfortunately, this template has no unit tests right now, so you'll have to write some of the boilerplate too. If you'd prefer not to, I'm also happy to merge this in since the change itself is pretty straightforward -- let me know.)

**Comment by nirav24** [comment]
@viswajithiii I was trying to figure out how to write a test. I will look at the referenced file and give it a shot.

**Comment by nirav24** [comment]
@viswajithiii I've added some tests. Would you please check whether that's how you'd like?

File: pkg/templates/serviceaccount/template_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+			Param: params.Params{
+				ServiceAccount: "non-default",
+			},
+			Diagnostics: map[string][]diagnostic.Diagnostic{
+				nonDefaultServiceAccount: {
+					{Message: "found matching serviceAccount (\"non-default\")"},
+				},
+				noAutoMountServiceAccountToken: {},
+			},
+			ExpectInstantiationError: false,
```
Comment (viswajithiii) [inline pkg/templates/serviceaccount/template_test.go:57]: Right now, we are testing:
1. matching service account + autoMount set to true explicitly => violation
2. non-matching service account + autoMount set to false explicitly => no violation

We should also test the other cases in the matrix, ie:
1. matching service account + autoMount set to `nil` => violation
2. matching service account + autoMount set to `false` explicitly => no violation 
3. non-matching service account + autoMount set to `true` explicitly => no violation
4. non-matching service account + autoMount set to `nil` => no violation

File: pkg/templates/serviceaccount/template_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		},
+		{
+			Param: params.Params{
+				ServiceAccount: "[a)", // Wrong Regex which should raise an error
```
Comment (viswajithiii) [inline pkg/templates/serviceaccount/template_test.go:61]: 👍

File: pkg/templates/serviceaccount/template_test.go (hunk 1 above)
Comment (nirav24) [reply pkg/templates/serviceaccount/template_test.go:57]: I will add other tests. Thanks for the suggestion. Appreciate it.

**Review by viswajithiii — APPROVED** [review]
Perfect. Thanks you so much @nirav24!

**Pull request merged**